src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,34 @@
-import React from 'react';
+import { FC } from 'react';
+
+import { SelectedType } from '../../types/SelectedType';
+import { useAppDispatch, useAppSelector } from '../../app/hook';
+import { filterSlice } from '../../features/filter';
+
+export const TodoFilter: FC = () => {
+  const dispatch = useAppDispatch();
+
+  const handleSelectedCondition = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    dispatch(filterSlice.actions.setFilter(event.target.value as SelectedType));
```

Comment:
Ensure that the `SelectedType` enum or type includes all the values used in the `<option>` elements to prevent runtime errors. Verify that `SelectedType` is correctly defined and imported.